apps/bsd/src/screens/LoadElectionScreen.tsx

```diff
@@ -74,9 +75,14 @@ const LoadElectionScreen: React.FC<Props> = ({
       await new Promise<void>((resolve, reject) => {
         reader.onload = async () => {
           try {
-            const election = JSON.parse(reader.result as string)
-            await patchConfig({ election })
-            setElection(election)
+            const electionData = reader.result as string
+            const electionDefinition = {
+              electionData,
+              electionHash: sha256(electionData),
+              election: JSON.parse(electionData),
+            }
+            await config.setElectionDefinition(electionDefinition)
+            setElectionDefinition(electionDefinition)
```

Comment:
This is the primary functional change at the point where we actually read the `election.json` file.